proof_of_play.go

```diff
@@ -83,26 +87,22 @@ func (p *proofOfPlay) Stop() {
 	close(p.retryQueue)
 }
 
-func (p *proofOfPlay) Expire(ad Ad) {
-	p.requests <- &PoPRequest{Ad: ad, Status: false}
+func (p *proofOfPlay) Expire(ad Ad) error {
+	req := &PoPRequest{Ad: ad, Status: false}
+	return p.expire(req)
 }
 
-func (p *proofOfPlay) Confirm(ad Ad, displayTime int64) {
-	p.requests <- &PoPRequest{Ad: ad, Status: true, DisplayTime: displayTime}
+func (p *proofOfPlay) Confirm(ad Ad, displayTime int64) error {
+	req := &PoPRequest{Ad: ad, Status: true, DisplayTime: displayTime}
+	return p.confirm(req)
 }
 
 func (p *proofOfPlay) start() {
 	for req := range p.requests {
 		if req.Status {
-			err := p.confirm(req.Ad, req.DisplayTime)
-			if err != nil {
-				p.processRequestFailure(req, err)
-			}
+			p.confirm(req)
```

Comment:
should we log errors here and on L105?

Reply:
Do you mean with events?

Reply:
no I was thinking regular logging because we're swallowing errors here but I forgot that we might not have logging available here. you can ignore this.